### PR TITLE
Prep 7.7.0 release

### LINKED ISSUE #1113: parsec should ignore unknown items, not abort
At least for global config (need to think about implications for suites...) This would make global config parsing forward-compatible.

### PR DESCRIPTION
**Pull request by hjoliver**
Updates to change log etc. for 7.7.0 release.

[update] close #1113

### CODE REVIEW

**Comment by hjoliver** [comment]
@matthewrmshin @sadielbartholomew - previously I've updated CHANGES.md and tagged the new release directly on master, but with non-trivial changes it should be done via a reviewed PR.  Additional things done here:
 * updated the CUG section on remote job host interaction (taken from my recent investigation of ssh-free job management)
 * addressed a few ancient quirks in the source tree: moved a few things into `etc` (e.g. got rid of the non-standard and badly named `dev` dir, used `usr/bin` for the main wrapper (like Rose), and updated docs and references to these locations appropriately.

**Comment by matthewrmshin** [comment]
Tests OK in my environment.

File: doc/src/cylc-user-guide/job-host.tex [hunk 1]
```diff
@@ -1,88 +1,356 @@
...
+SSH.\footnote{Cylc used to run bare shell expressions over SSH, which required
+a bash shell and made whitelisting difficult.} This section shows exactly what
+cylc does on and to remote job hosts.
 
-This appendix shows transcripts of all \lstinline=ssh=, \lstinline=scp=, and
-\lstinline=rsync= invocations from a Cylc suite called \lstinline=test= started by
-\lstinline=rose suite-run= on \lstinline=vagrant@localhost=. The suite
-submits a single task called \lstinline=foo= to \lstinline=hobo@otherhost=.
-The operator polls then kills the task, then shuts the suite down.
+Security-conscious sites may want to restrict access to job hosts with SSH
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/job-host.tex]: Can we tone this down slightly? It is not that other sites are not security conscious. It is that sites may have different security audit requirements.

File: doc/src/cylc-user-guide/job-host.tex [hunk 2]
```diff
@@ -1,88 +1,356 @@
...
+   # (internal rsync, back from hobo@wizard)
+   rsync --server -logDtpre.iLsfx --timeout=1800 . cylc-run/suitex
+\end{lstlisting}
+
+\vspace{5mm}
+Result:
+\lstset{language=sh}
+{\scriptsize
+\dirtree{%
+.1 /home/hjoliver/cylc-run/suitex.
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/job-host.tex]: Use `~/` to make it more general?

File: doc/src/cylc-user-guide/job-host.tex [hunk 3]
```diff
@@ -1,88 +1,356 @@
...
+# cylc remote-init: install suite service directory
+ssh -oBatchMode=yes -oConnectTimeout=10 hobo@wizard env CYLC_VERSION=7.6.x /opt/bin/cylc remote-init '066592b1-4525-48b5-b86e-da06eb2380d9' '$HOME/cylc-run/suitex'
+\end{lstlisting}
 
-# vagrant@localhost: submit job foo.1 (cycle point 1, try 01) on job host
-ssh -oBatchMode=yes -oConnectTimeout=10 hobo@otherhost env CYLC_VERSION=7.3.0 cylc jobs-submit '--remote-mode' '--' '$HOME/cylc-run/test/log/job' '1/foo/01'
+Result:
+{\scriptsize
+\dirtree{%
+.1 /home/hjoliver/cylc-run/suitex.
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/job-host.tex]: Use `~/`?

File: doc/src/cylc-user-guide/job-host.tex [hunk 4]
```diff
@@ -1,88 +1,356 @@
...
+# cylc jobs-submit: submit two jobs
+ssh -oBatchMode=yes -oConnectTimeout=10 hobo@wizard env CYLC_VERSION=7.6.x /opt/bin/cylc jobs-submit '--remote-mode' '--' '$HOME/cylc-run/suitex/log/job' '1/remote-a/01' '1/remote-b/01'
+\end{lstlisting}
 
-# (INTERNAL RSYNC COMMAND) vagrant@locahost
-ssh -oBatchMode=yes -oConnectTimeout=10 -l hobo otherhost rsync --server --sender -logDtpre.iLs . $HOME/cylc-run/test/log/job/
+Result:
+{\scriptsize
+\dirtree{%
+.1 /home/hjoliver/cylc-run/suitex.
```
Comment (matthewrmshin) [inline doc/src/cylc-user-guide/job-host.tex]: Use `~/`?

File: tests/tutorial/test_header_tutorial [hunk 5]
```diff
@@ -39,7 +39,11 @@ EOF
     # *** WARNING: ASSUMING ALL THE SUITES HAVE A 12-HOURLY CYCLE
     # (this is just for consistency - reference tests actually take
     # initial and final cycles from reference log, not the suite def).
-    perl -pi -e "s/(initial cycle point\s*=)\s*.*/\1 20130808T00/g" suite.rc
-    perl -pi -e "s/(final cycle point\s*=)\s*.*/\1 20130808T12/g" suite.rc
-
+    if grep "cycling mode = integer" suite.rc > /dev/null; then
```
Comment (matthewrmshin) [inline tests/tutorial/test_header_tutorial]: `grep -q ...`?

**Comment by hjoliver** [comment]
@matthewrmshin - feedback all addressed.  Over 'n out for today.

**Comment by matthewrmshin** [comment]
See also https://github.com/cylc/cylc/issues/1113#issuecomment-310488417 for issue with `global.rc`.

**Comment by matthewrmshin** [comment]
(I think it is worth moving site global to `etc/global.rc`, with back compat logic to also search for the file under `conf/global.rc`.)

**Comment by hjoliver** [comment]
`cylc message` help and docs updated.

**Comment by hjoliver** [comment]
> (I think it is worth moving site global to etc/global.rc, with back compat logic to also search for the file under conf/global.rc.)

OK, I've done that now.  Note I've dropped the deprecated `site.rc` and `user.rc` filename variants - they are years old now, no one should be using them.

**Comment by hjoliver** [comment]
> See also #1113 (comment) for issue with global.rc.

Well, you asked for it.  I've had a crack at this now. My new solution mimics the site update method (keep it version specific):

```
~/.cylc/x.y.z/global.rc
```
where x.y.z is cylc version (and is truncated beyond z, for use in git clones); falling back to `~/.cylc/global.rc`.

New file `etc/global.rc.eg` explains the rationale.  I think this is better than what we have now, because it allows users to work with multiple versions at once, which may not be possible at the moment depending on your global settings.  On the downside, they have to know to copy over the global config file before using a new version.

Thoughts?

**Comment by matthewrmshin** [comment]
3 tests broken by directory restructure. Need to make this change:

```diff
diff --git a/lib/cylc/version.py b/lib/cylc/version.py
index 8c339bd0b..26a91190d 100644
--- a/lib/cylc/version.py
+++ b/lib/cylc/version.py
@@ -34,8 +34,8 @@ def _get_cylc_version():
         # We're running in a cylc git repository, so dynamically determine
         # the cylc version string.  Enclose the path in quotes to handle
         # avoid failure when cylc_dir contains spaces.
-        is_ok, outlines = run_get_stdout(
-            '"%s"' % os.path.join(cylc_dir, "etc", "bin", "get-repo-version"))
+        is_ok, outlines = run_get_stdout('"%s"' % os.path.join(
+            cylc_dir, "etc", "dev-bin", "get-repo-version"))
         if is_ok and outlines:
             return outlines[0]
         else:
```

**Comment by matthewrmshin** [comment]
I think we should close #1113 after this - the intended requirement is met by this solution.

**Comment by hjoliver** [comment]
(Marked #1113 to close when this is merged)

**Comment by hjoliver** [comment]
Final commit makes the regenerated tutorial reference tests non-detaching (as reference tests are supposed to be ... but unfortunately `suite_run_ok cylc run foo --reference-test` just passes if `--no-detach` is omitted).

That should be it ... hopefully I can make the release tomorrow.

**Comment by matthewrmshin** [comment]
@oliver-sanders please do a final sanity check.

File: doc/src/cylc-user-guide/cug.tex [hunk 6]
```diff
@@ -3779,7 +3794,7 @@ \subsubsection{Trigger Types}
 message. Other tasks can trigger off the externally triggered task as required,
 of course.
 
-\lstinline=$CYLC_DIR/examples/satellite/ext-triggers/suite.rc= is a working
+\lstinline=<cylc-dir>/etc/examples/satellite/ext-triggers/suite.rc= is a working
```
Comment (oliver-sanders) [inline doc/src/cylc-user-guide/cug.tex]: I like the `<substitution>` syntax, in the rose documentation we now have a pygments lexer for highlighting these in code blocks e.g: http://metomi.github.io/rose/doc/html/cheat-sheet.html

Comment (hjoliver) [reply doc/src/cylc-user-guide/cug.tex]: Nice, looking foward to going cylc docs going this way...

**Pull request merged**